dds/DCPS/RTPS/TypeLookup.cpp

```diff
@@ -0,0 +1,326 @@
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#include "TypeLookup.h"
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+
+namespace DCPS {
+
+  void serialized_size(const Encoding& encoding, size_t& size,
+    const XTypes::TypeLookup_getTypes_In& stru)
+  {
+    serialized_size(encoding, size, stru.type_ids);
+  }
+
+  bool operator<<(Serializer& strm, const XTypes::TypeLookup_getTypes_In& stru)
+  {
+    return strm << stru.type_ids;
+  }
+
+  bool operator>>(Serializer& strm, XTypes::TypeLookup_getTypes_In& stru)
+  {
+    return strm >> stru.type_ids;
+  }
+
+
+  void serialized_size(const Encoding& encoding, size_t& size,
+    const XTypes::TypeLookup_getTypes_Out& stru)
+  {
+    serialized_size(encoding, size, stru.types);
+    serialized_size(encoding, size, stru.complete_to_minimal);
+  }
+
+  bool operator<<(Serializer& strm, const XTypes::TypeLookup_getTypes_Out& stru)
+  {
+    return (strm << stru.types)
+      && (strm << stru.complete_to_minimal);
+  }
+
+  bool operator>>(Serializer& strm, XTypes::TypeLookup_getTypes_Out& stru)
+  {
+    return (strm >> stru.types)
+      && (strm >> stru.complete_to_minimal);
+  }
+
+
+  void serialized_size(const Encoding& encoding, size_t& size,
+    const XTypes::TypeLookup_getTypes_Result& stru)
```

Comment:
This is an IDL union, so `serialized_size` and the two operators need to follow the pattern for unions.